drivers/dma/dma_iproc_pax.h

```diff
@@ -0,0 +1,524 @@
+	(!((addr) & ((0x1 << RING_CMPL_ALIGN_ORDER) - 1)))
+
+/* RM descriptor width: 8 bytes */
+#define PAX_DMA_RM_DESC_BDWIDTH		8
+/* completion msg desc takes 1 BD */
+#define PAX_DMA_CMPL_DESC_SIZE		PAX_DMA_RM_DESC_BDWIDTH
+/* Next table desc takes 1 BD */
+#define PAX_DMA_NEXT_TBL_DESC_SIZE	PAX_DMA_RM_DESC_BDWIDTH
+/* Header desc takes 1 BD */
+#define PAX_DMA_HEADER_DESC_SIZE	PAX_DMA_RM_DESC_BDWIDTH
+/* Total BDs in ring: 4K/8bytes = 512 BDs */
+#define PAX_DMA_RM_RING_BD_COUNT	(PAX_DMA_RM_DESC_RING_SIZE / \
+					 PAX_DMA_RM_DESC_BDWIDTH)
+
+/* Initial RM header is first BD in ring */
+#define PAX_DMA_HEADER_INDEX		0
+#define PAX_DMA_HEADER_ADDR(_ring)	(void *)((uintptr_t)(_ring) + \
+	PAX_DMA_HEADER_INDEX * PAX_DMA_RM_DESC_BDWIDTH)
+
+/* NEXT TABLE desc offset is last BD in ring */
+#define PAX_DMA_NEXT_TBL_INDEX		(PAX_DMA_RM_RING_BD_COUNT - 1)
+#define PAX_DMA_NEXT_TBL_ADDR(_ring)	(void *)((uintptr_t)(_ring) + \
+	PAX_DMA_NEXT_TBL_INDEX * PAX_DMA_RM_DESC_BDWIDTH)
+
+/* DMA transfers supported from 4 bytes thru 16M, size aligned to 4 bytes */
+#define PAX_DMA_MIN_SIZE	4
+#define PAX_DMA_MAX_SIZE	(16 * 1024 * 1024)
+
+/* Host and Card address need 4-byte alignment */
+#define PAX_DMA_ADDR_ALIGN 4
+
+#define RM_RING_REG(_pd, _r, _write_ptr) \
+	((_pd)->ring[_r].ring_base + (_write_ptr))
+#define RM_COMM_REG(_pd, _write_ptr) ((_pd)->rm_comm_base + (_write_ptr))
+#define PAX_DMA_REG(_pd, _write_ptr) ((_pd)->dma_base + (_write_ptr))
+
+#define PAX_DMA_MAX_CMPL_COUNT		1024
+#define PAX_DMA_LAST_CMPL_IDX		(PAX_DMA_MAX_CMPL_COUNT - 1)
+
+#define PAX_DMA_RING_ALIGN		BIT(PAX_DMA_RING_CMPL_ALIGN_ORDER)
+/* num of completions received, circular buffer */
+#define PAX_DMA_GET_CMPL_COUNT(wptr, rptr) (((wptr) >= (rptr)) ? \
+	((wptr) - (rptr)) : (PAX_DMA_MAX_CMPL_COUNT - (rptr) + (wptr)))
+
+/* location of current cmpl pkt, take care of pointer wrap-around */
+#define PAX_DMA_CURR_CMPL_IDX(wptr)	\
+	(((wptr) == 0) ? PAX_DMA_LAST_CMPL_IDX : (wptr) - 1)
+
+/* Timeout (milliseconds) for completion alert in interrupt mode */
+#define PAX_DMA_TIMEOUT			10000
+
+/* TODO: add macro to enable memory barrier */
+#define dma_mb()
+
+/* Max polling cycles for completion wait, >= 1 second */
+#define PAX_DMA_MAX_POLL_WAIT		1000000
+/* Max polling cycles for posted write sync >= 1 second */
+#define PAX_DMA_MAX_SYNC_WAIT		1000000
+
+enum ring_idx {
+	PAX_DMA_RING0 = 0,
+	PAX_DMA_RING1,
+	PAX_DMA_RING2,
+	PAX_DMA_RING3,
+	PAX_DMA_RINGS_MAX
+};
+
+/*
+ * DMA direction
+ */
+enum pax_dma_dir {
+	CARD_TO_HOST = 0x1,
+	HOST_TO_CARD = 0x2
+};
+
+/* Completion packet */
+struct cmpl_pkt {
+	uint64_t opq : 16; /*pkt_id 15:0*/
+	uint64_t res : 16; /*reserved 16:31*/
+	uint64_t dma_status : 16; /*PAX DMA status 32:47*/
+	uint64_t ae_num : 6; /*RM status[47:53] processing AE number */
+	uint64_t rm_status : 10; /*RM status[54:63] completion/timeout status*/
+} __attribute__ ((__packed__));
+
+/* Driver internal structures */
+
+struct dma_iproc_pax_addr64 {
```

Comment:
I know I asked for prefixing types/defines to not pollute global namespace, "dma_iproc_pax" is a bit on the verbose side and "pax" would probably suffice, but I'm not going to ask you to change it.

Reply:
Followed naming convention lookin at other dma driver's, I understand its bit verbose.  For register definition macros, as explained earlier register name + bit field name together is becoming too lengthy if I add some common prefix.

But as such register macros are classified according to the sub blocks within Ring manager like: RING, RM_COMM, RM_AEx. Other driver specific macros are named with prefix PAX_DMA.